Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python:3.12.3-alpine as base
+    RUN pip install poetry
+    WORKDIR /app
+    ENTRYPOINT ["poetry", "run", "flask", "run", "--host=0.0.0.0"]
```

Comment:
(Even if it's functionally identical, I do prefer the ENTRYPOINT at the end of the stage, since conceptually its the last thing that will actually execute)